### PR TITLE
Lf 2502/hotfix in temperature component due to database migration changes

### PR DESCRIPTION
**Pull request by anant00729**
To test,

1. Add sensor readings to the sensor_reading table. 
2. check if you can see sensor readings.

### CODE REVIEW

**Review by jagDhillon077 — APPROVED** [review]
LGTM - temperature showing up in both metric and imperial without any errors

File: packages/api/src/models/sensorReadingModel.js [hunk 1]
```diff
@@ -87,6 +87,7 @@ class SensorReading extends Model {
     return await SensorReading.query()
       .select('*')
       .joinRaw('JOIN sensor ON sensor_reading.location_id::uuid = sensor.location_id')
+      .joinRaw('JOIN location ON location.location_id::uuid = sensor.location_id::uuid')
```
Comment (Cali0707) [inline packages/api/src/models/sensorReadingModel.js:90]: As far as I know, all of the columns for location_id are already type `uuid`, so I believe we do not need to cast them to type `uuid`

File: packages/api/src/routes/sensorRoute.js [hunk 2]
```diff
@@ -46,5 +46,5 @@ router.post('/reading/invalidate', SensorController.invalidateReadings);
 router.post('/unclaim', SensorController.retireSensor);
 router.get('/:location_id/reading_type', SensorController.getSensorReadingTypes);
 router.get('/partner/:partner_id/brand_name', SensorController.getBrandName);
-router.get('/reading/visualization', SensorController.getAllSensorReadingsByLocationIds);
+router.post('/reading/visualization', SensorController.getAllSensorReadingsByLocationIds);
```
Comment (Cali0707) [inline packages/api/src/routes/sensorRoute.js:49]: Is there a reason why this is being switched to a POST request? If I'm understanding what's going on, this endpoint retrieves the sensor readings, so it should be a GET request?

**Review by Cali0707 — CHANGES_REQUESTED** [review]

File: packages/api/src/routes/sensorRoute.js (hunk 2 above)
Comment (anant00729) [reply packages/api/src/routes/sensorRoute.js:49]: it was POST initially, I am using the post method to send the data in the request body. I got this new change from database migration PR that I had to again change it from GET to POST in this PR.

File: packages/api/src/models/sensorReadingModel.js (hunk 1 above)
Comment (anant00729) [reply packages/api/src/models/sensorReadingModel.js]: Done

File: packages/api/src/routes/sensorRoute.js (hunk 2 above)
Comment (Cali0707) [reply packages/api/src/routes/sensorRoute.js:49]: 👍

File: packages/api/src/models/sensorReadingModel.js [hunk 3]
```diff
@@ -87,6 +87,7 @@ class SensorReading extends Model {
     return await SensorReading.query()
       .select('*')
       .joinRaw('JOIN sensor ON sensor_reading.location_id::uuid = sensor.location_id')
```
Comment (Cali0707) [inline packages/api/src/models/sensorReadingModel.js:89]:
```suggestion
      .joinRaw('JOIN sensor ON sensor_reading.location_id = sensor.location_id')
```
`sensor_reading.location_id` is type `uuid`, no reason for this to be type casted either

**Review by Cali0707 — CHANGES_REQUESTED** [review]

File: packages/api/src/models/sensorReadingModel.js (hunk 3 above)
Comment (anant00729) [reply packages/api/src/models/sensorReadingModel.js]: corrected that as well. Thanks!

**Review by Cali0707 — APPROVED** [review]
Looks good to me!

**Pull request merged**